### PR TITLE
#22: Cover with tests CustomUser model

### LINKED ISSUE #57: #22: Cover with tests CustomUser model
1) Read guideline [here](https://gist.github.com/iamnovichek/801ac395a06a7d8b04b022084ace684d)
2) Write tests for `CustomUser` class from `tracker/models.py` in `tracker/tests/test_models.py`
3) Provide some screenshots to prove that your tests work

### PR DESCRIPTION
**Pull request by cryptogru725**
Close #57

### CODE REVIEW

File: tracker/tests/test_models.py [hunk 1]
```diff
@@ -54,3 +54,66 @@ def test_create_user_with_invalid_email(self):
...
+    def test_str_method(self):
+        """Test the __str__ method of the CustomUser model."""
+        user = CustomUser.objects.create_user(
+            email=self.email,
+            is_active=self.is_active,
+            is_admin=self.is_admin,
+            role=self.role,
+        )
+        
+        self.assertEqual(CustomUser.__str__(user), self.email)
```
Comment (iamnovichek) [inline tracker/tests/test_models.py]: pls use just `str(user)`

File: tracker/tests/test_models.py [hunk 2]
```diff
@@ -54,3 +54,66 @@ def test_create_user_with_invalid_email(self):
             )
 
         self.assertEqual(str(context.exception), "Invalid email format")
+
+class TestCustomUser(TestCase):
+    def setUp(self):
+        """Set up test data."""
+        self.email = fake.email()
+        self.is_active = fake.boolean()
```
Comment (iamnovichek) [inline tracker/tests/test_models.py]: Pls set it to `True`

File: tracker/tests/test_models.py [hunk 3]
```diff
@@ -54,3 +54,66 @@ def test_create_user_with_invalid_email(self):
...
+        self.is_admin = fake.boolean()
```
Comment (iamnovichek) [inline tracker/tests/test_models.py]: Same

File: tracker/tests/test_models.py [hunk 4]
```diff
@@ -54,3 +54,66 @@ def test_create_user_with_invalid_email(self):
...
+    def test_is_staff(self):
+        """Test the is_staff property of the CustomUser model."""
+        user = CustomUser.objects.create_user(
+            email=self.email,
+            is_active=self.is_active,
+            is_admin=self.is_admin,
+            role=self.role,
+        )
+
+        self.assertEqual(CustomUser.is_staff(user), self.is_admin)
```
Comment (iamnovichek) [inline tracker/tests/test_models.py]: pls use `user.is_staff`

File: tracker/tests/test_models.py [hunk 5]
```diff
@@ -54,3 +54,66 @@ def test_create_user_with_invalid_email(self):
...
+    def test_is_project_lead(self):
+        """Test the is_project_lead method of the CustomUser model."""
+        user = CustomUser.objects.create_user(
+            email=self.email,
+            is_active=self.is_active,
+            is_admin=self.is_admin,
+            role=self.role,
+        )
+
+        self.assertEqual(CustomUser.is_project_lead(user), self.role == Roles.PROJECT_LEAD)
```
Comment (iamnovichek) [inline tracker/tests/test_models.py]: Same here

**Review by iamnovichek — CHANGES_REQUESTED** [review]

File: tracker/tests/test_models.py [hunk 6]
```diff
@@ -54,3 +54,66 @@ def test_create_user_with_invalid_email(self):
...
+    def test_is_staff(self):
+        """Test the is_staff property of the CustomUser model."""
+        user = CustomUser.objects.create_user(
+            email=self.email,
+            is_active=self.is_active,
+            is_admin=self.is_admin,
+            role=self.role,
+        )
+
+        self.assertEqual(user.is_staff(), self.is_admin)
```
Comment (iamnovichek) [inline tracker/tests/test_models.py]: Use just `user.is_staff`

File: tracker/tests/test_models.py [hunk 7]
```diff
@@ -54,3 +54,66 @@ def test_create_user_with_invalid_email(self):
...
+    def test_is_project_lead(self):
+        """Test the is_project_lead method of the CustomUser model."""
+        user = CustomUser.objects.create_user(
+            email=self.email,
+            is_active=self.is_active,
+            is_admin=self.is_admin,
+            role=self.role,
+        )
+
+        self.assertEqual(user.is_project_lead(), self.role == Roles.PROJECT_LEAD)
```
Comment (iamnovichek) [inline tracker/tests/test_models.py]: Same. You don't need to call a property

**Review by iamnovichek — CHANGES_REQUESTED** [review]

**Review by iamnovichek — APPROVED** [review]

**Comment by iamnovichek** [comment]
@cryptogru812 tobi pyzda

**Comment by cryptogru725** [comment]
??? @iamnovichek

**Pull request merged**